### ISSUE TITLE
Emulated vs. real (SD card) external storage path

### ISSUE THREAD

**Issue by transistorgrab** [nguillaumin/osmtracker-android#49]
I use a Sony Xperia Tablet for tracking. It runs currently Android 5.0.2.
When I set a custom path it always stores in the `/sdcard` path (emulated SD card in internal memory) and not on `/sdcard1` which is the external/additional actual SD card.
So if I put a path `/sdcard1/osmtracker` in the settings it acutally stores the data then in `/sdcard/sdcard1/osmtracker`. (took me some time to find my tracked data...)
Is there any way to enable OSMtracker to write on the acutal SD card?
((On my (rooted) smartphone running Android 4.1.2 this is no issue... but this device is not my main tracking device due to age, accuracy and speed.))
Also I think it would be a good thing to be able to set the path by browsing to the desired directory than needing to type in the path. Then OSMtracker maybe should check if it can actually access the desired path.

**Comment by nguillaumin** [comment]
Hi,

I think it's a well known problem of the Android SDK, there's is no "official" way to obtain the path to the actual SD card, because it's abstracted away by the [external storage](http://developer.android.com/reference/android/os/Environment.html#getExternalStorageDirectory()) in the API.

Android 4.4 has [introduced improvements](https://source.android.com/devices/storage/#multiple-external-storage-devices) for this but I'm not sure we could use them in OSMTracker because they are usable only to store "media" files, rather than having a single root folder like `/osmtracker` that we can organise as we please. In any case that would require Android 4.4 which is not ideal.

Another option could be to stop using the "external storage" abstraction and just use the path as given in the settings, but that will cause all sort of problems for users who don't know what the path to the SD card and/or internal storage is.

So to sum up, there's no easy solution. I'd happily take a patch for that, perhaps for the latter by making it an option (e.g. a toggle "use path as is" or "use path relative to external storage"), but I won't be able to work on that on my own.

Regarding the file browser that's a nice idea, but similarly I'm not willing to implement a file browser in OSMTracker but will happily take a patch for it.

Cheers.

**Comment by transistorgrab** [comment]
Thanks for the reply and thorough explanation. Since I'm not a software developer I think I can not come up with a solution though… 😉
